kratos/geometries/geometry_shape_function_container.h

```diff
@@ -326,23 +326,21 @@ class GeometryShapeFunctionContainer
         IndexType DerivativeOrderIndex,
         IntegrationMethod ThisMethod) const
     {
-        if (DerivativeOrderIndex > 1)
-        {
-            KRATOS_DEBUG_ERROR_IF(mShapeFunctionsDerivatives[ThisMethod][IntegrationPointIndex].size() > (DerivativeOrderIndex - 2))
```

Comment:
does this leave a warning?

Because if we move
KRATOS_ERROR_IF(DerivativeOrderIndex == 0)
            << "Shape functions cannot be accessed through ShapeFunctionDerivatives()" << std::endl;
to the end we would reduce it with one if in the general cases.

Otherwise I would like to make a debug error out of it.

Reply:
Yep basically what I did is reorder the code, but the behavior should be the same:
- The `KRATOS_ERROR_IF(DerivativeOrderIndex == 0)` is moved to the top because is independent on any if, so it should not matter if its triggered at the end o at the beginning, if it reached that code it has to reach it without evaluating oany of the other ifs.
- The `if (DerivativeOrderIndex > 1) {KRATOS_DEBUG_ERROR_IF(mShape...` is moved after the first `if` so we can ensure that `DerivativeOrderIndex != 1` and `DerivativeOrderIndex > 0` because is a `IndexType` so i removed the if because will be always be true (Otherwise it would have exit the function in the first conditional), and the default return is reached using the same criteria so its as having:

```C++
if (DerivativeOrderIndex > 1) {
    KRATOS_DEBUG_ERROR_IF(mShapeFunctionsDerivatives[ThisMethod][IntegrationPointIndex].size() > (DerivativeOrderIndex - 2)) << "Not enough derivatives within geometry_shape_function_container." << std::endl;

    return mShapeFunctionsDerivatives[ThisMethod][IntegrationPointIndex][DerivativeOrderIndex - 2];
}
```

Without the if, so the warning of non default return value is gone